mkosi/config.py

```diff
@@ -192,6 +192,22 @@ def config_match_enum(dest: str, value: str, namespace: argparse.Namespace) -> b
 
     return config_match_enum
 
+def config_make_list_matcher(type: Type[enum.Enum], delimiter: str) -> ConfigMatchCallback:
```

Comment:
Similar to `config_make_list_parser()`, this should take `match` function that does the actual match for a single element in the list.

Reply:
I cannot fathom how these Callable things work, even after reading the Python docs, and the prototype for `ConfigParseCallback`  has a different prototype to `ConfigMatchCallback` so I'm grasping at straws. Since the code works as-is, and with the following possible revision, I'm getting no help from the interpreter either. Give me a strongly typed compiled language any day - help!
```
# prototype: ConfigMatchCallback = Callable[[str, str, argparse.Namespace], bool]
def config_make_list_matcher(type: Type[enum.Enum], delimiter: str, parse: Callable[[str], Any] = bool) -> ConfigMatchCallback:
    def config_match_list(dest: str, value: str, namespace: argparse.Namespace) -> bool:
```

Reply:
`Callable` is just how to encode a function in the type system. The first argument to `Callable` is the list of types of the arguments (only positional arguments are supported), the second is the return type. 

The interpreter is of little help here, but you can run a type checker, mainly `mypy`, but also `pyright` and others on it. It's not much different from a statically compiled language then.